src/Invictus.Testing/LogicAppException.cs

```diff
@@ -0,0 +1,127 @@
+using GuardNet;
+
+namespace Invictus.Testing
+{
+    /// <summary>
+    /// Thrown when a problem occurs during interaction with a logic app running in Azure.
+    /// </summary>
+    [Serializable]
+    public class LogicAppException : Exception
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogicAppException"/> class.
+        /// </summary>
+        public LogicAppException()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogicAppException"/> class.
+        /// </summary>
+        /// <param name="message">The message that describes the exception.</param>
+        public LogicAppException(string message) : base(message)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogicAppException"/> class.
+        /// </summary>
+        /// <param name="message">The message that describes the exception.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception</param>
+        public LogicAppException(string message, Exception innerException) : base(message, innerException)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogicAppException"/> class.
+        /// </summary>
+        /// <param name="message">The message that describes the exception.</param>
+        /// <param name="logicAppName">The name of the logic app resource running in Azure.</param>
+        /// <param name="resourceGroup">The resource group where the logic app is located.</param>
+        /// <param name="subscriptionId">The ID that identifies the subscription on Azure.</param>
+        public LogicAppException(
+            string message,
+            string logicAppName,
+            string resourceGroup,
+            string subscriptionId) : base(message)
+        {
+            Guard.NotNullOrWhitespace(logicAppName, nameof(logicAppName));
+            Guard.NotNullOrWhitespace(resourceGroup, nameof(resourceGroup));
+            Guard.NotNullOrWhitespace(subscriptionId, nameof(subscriptionId));
+
+            LogicAppName = logicAppName;
+            ResourceGroup = resourceGroup;
+            SubscriptionId = subscriptionId;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogicAppException"/> class.
+        /// </summary>
+        /// <param name="message">The message that describes the exception.</param>
+        /// <param name="logicAppName">The name of the logic app resource running in Azure.</param>
+        /// <param name="resourceGroup">The resource group where the logic app is located.</param>
+        /// <param name="subscriptionId">The ID that identifies the subscription on Azure.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception</param>
+        public LogicAppException(
```

Comment:
We should always order from least to most specific meaning - subscription, resource group, logic app name, message

Reply:
Yes, I see, but if we have a sole-message ctor, it seems like it's more discoverable if we start with the message.

Reply:
Also, since it's all strings, it's easy to miss if we have ctors with changed positions.

Reply:
We haven't shipped it yet so now is the time to re-order. Personally I think message should always be at the end because they are long and if folks don't add new lines they won't even notice that other (more valuable) info is passed.

For me message is even optional if we provide the detailed info since the ctor can provide the message based on that.

Reply:
Changed message as later param.